cluster-autoscaler/simulator/predicatechecker/schedulerbased.go

```diff
@@ -104,8 +106,16 @@ func (p *SchedulerBasedPredicateChecker) FitsAnyNodeMatching(clusterSnapshot clu
 	p.delegatingSharedLister.UpdateDelegate(clusterSnapshot)
 	defer p.delegatingSharedLister.ResetDelegate()
 
+	fwk, err := p.frameworkForPod(pod)
+	if err != nil {
+		// This shouldn't happen, because we only accept for scheduling the pods
```

Comment:
What does that mean? Is there an admission controller built-into kubernetes that prevents users from creating pods with arbitrary names. What if someone has a custom scheduler that is not a profile of kubernetes scheduler, but a completely custom binary? Today we would still scale-up for them - potentially using different criteria, but maybe they've already accounted for that. Isn't this a breaking change?

Wouldn't it be more backward compatible to return default scheduler if scheduler name is not found?

Reply:
> Is there an admission controller built-into kubernetes that prevents users from creating pods with arbitrary name

No.

> What if someone has a custom scheduler that is not a profile of kubernetes scheduler, but a completely custom binary? Today we would still scale-up for them

Also cases where people have multiple kube-scheduler profiles (for example, just to have different scoring strategies). These users would suddenly be broken unless they create all the same profiles for their cluster autoscaler instance.

> Wouldn't it be more backward compatible to return default scheduler if scheduler name is not found?

I agree. Maybe just the first profile from the list.

Reply:
ok, I made the modifications as requested.